proxmoxtf/resource/vm.go

```diff
@@ -4217,6 +4217,11 @@ func vmReadCustom(
 		}
 
 		disk[mkResourceVirtualEnvironmentVMDiskInterface] = di
+
+		if dd.Size == nil {
+			dd.Size = new(types.DiskSize)
+		}
```

Comment:
Thanks @Jayden-Lind, that's a good catch!

This change definitely fixes the issue for your use case. However, the func `getDiskInfo` that populates the `CustomStorageDevice` struct is used in a few more places. 
So I would suggest to move this `if ...` statement over there, put it somewhere between [these 2 lines](https://github.com/bpg/terraform-provider-proxmox/blob/main/proxmoxtf/resource/vm.go#L6409-L6410).

Reply:
Thanks @bpg, will check it out tonight and update it 😄

Reply:
Updated @bpg